### PR TITLE
Deprecate demisto-api-* commands

### PR DESCRIPTION
**Pull request by michal-dagan**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [link to the issue](https://jira-dc.paloaltonetworks.com/browse/CIAC-9102)
content-docs pr: https://github.com/demisto/content-docs/pull/1440

## Description
Deprecate all the demisto-api-* commands.
Replace their usage in OOB scripts and playbooks.

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/DemistoRESTAPI/ReleaseNotes/1_3_41.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
+
```
Comment (michal-dagan) [inline Packs/DemistoRESTAPI/ReleaseNotes/1_3_41.md:1]: @ShirleyDenkberg, please review this file, and if anything needs to be modified, I'll update the others

File: Packs/DemistoRESTAPI/ReleaseNotes/1_3_41.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+##### Core REST API
+
+- Command ***demisto-api-multipart*** is deprecated. Use core-api-multipart instead.
+- Command ***demisto-delete-incidents*** is deprecated. Use core-delete-incidents instead.
+- Command ***demisto-api-post*** is deprecated. Use core-api-post instead.
+- Command ***demisto-api-put*** is deprecated. Use core-api-put instead.
+- Command ***demisto-api-get*** is deprecated. Use core-api-get instead.
+- Command ***demisto-api-delete*** is deprecated. Use core-api-delete instead.
+- Command ***demisto-api-download*** is deprecated. Use core-api-download instead.
+- Command ***demisto-api-install-packs*** is deprecated. Use core-api-install-packs instead.
```
Comment (ShirleyDenkberg) [inline Packs/DemistoRESTAPI/ReleaseNotes/1_3_41.md:13]:
```suggestion
- Command ***demisto-api-multipart*** is deprecated. Use ***core-api-multipart*** instead.
- Command ***demisto-delete-incidents*** is deprecated. Use ***core-delete-incidents*** instead.
- Command ***demisto-api-post*** is deprecated. Use ***core-api-post*** instead.
- Command ***demisto-api-put*** is deprecated. Use ***core-api-put*** instead.
- Command ***demisto-api-get*** is deprecated. Use ***core-api-get*** instead.
- Command ***demisto-api-delete*** is deprecated. Use ***core-api-delete*** instead.
- Command ***demisto-api-download*** is deprecated. Use ***core-api-download*** instead.
- Command ***demisto-api-install-packs*** is deprecated. Use ***core-api-install-packs*** instead.
```

File: Packs/DemistoRESTAPI/ReleaseNotes/1_3_41.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+#### Scripts
+
+##### DemistoGetIncidentTasksByState
+
+- Deprecated ***demisto-api-**** commands and replaced with ***core-api-**** commands.
```
Comment (ShirleyDenkberg) [inline Packs/DemistoRESTAPI/ReleaseNotes/1_3_41.md:23]:
```suggestion
Deprecated ***demisto-api-**** commands and replaced with ***core-api-**** commands.
```

File: Packs/DemistoRESTAPI/ReleaseNotes/1_3_41.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+##### DemistoUploadFileToIncident
+
+- Updated the Docker image to: *demisto/python:2.7.18.76921*.
+- Deprecated ***demisto-api-**** commands and replaced with ***core-api-**** commands.
+##### DemistoLinkIncidents
+
+- Deprecated ***demisto-api-**** commands and replaced with ***core-api-**** commands.
```
Comment (ShirleyDenkberg) [inline Packs/DemistoRESTAPI/ReleaseNotes/1_3_41.md:31]:
```suggestion
Deprecated ***demisto-api-**** commands and replaced with ***core-api-**** commands.
```

File: Packs/DemistoRESTAPI/ReleaseNotes/1_3_41.md [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+##### DemistoLeaveAllInvestigations
+
+- Deprecated ***demisto-api-**** commands and replaced with ***core-api-**** commands.
```
Comment (ShirleyDenkberg) [inline Packs/DemistoRESTAPI/ReleaseNotes/1_3_41.md:34]:
```suggestion
Deprecated ***demisto-api-**** commands and replaced with ***core-api-**** commands.
```

File: Packs/DemistoRESTAPI/ReleaseNotes/1_3_41.md [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+##### DemistoUploadFileV2
+
+- Updated the Docker image to: *demisto/python3:3.10.13.83255*.
+- Deprecated ***demisto-api-**** commands and replaced with ***core-api-**** commands.
+##### DemistoLogsBundle
+
+- Deprecated ***demisto-api-**** commands and replaced with ***core-api-**** commands.
```
Comment (ShirleyDenkberg) [inline Packs/DemistoRESTAPI/ReleaseNotes/1_3_41.md:42]:
```suggestion
Deprecated ***demisto-api-**** commands and replaced with ***core-api-**** commands.
```

File: Packs/DemistoRESTAPI/ReleaseNotes/1_3_41.md [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+##### DemistoSendInvite
+
+- Deprecated ***demisto-api-**** commands and replaced with ***core-api-**** commands.
+##### DemistoCreateList
+
+- Deprecated ***demisto-api-**** commands and replaced with ***core-api-**** commands.
+##### GetIncidentTasksByState
+
+- Deprecated ***demisto-api-**** commands and replaced with ***core-api-**** commands.
```
Comment (ShirleyDenkberg) [inline Packs/DemistoRESTAPI/ReleaseNotes/1_3_41.md:56]:
```suggestion
Deprecated ***demisto-api-**** commands and replaced with ***core-api-**** commands.
##### DemistoCreateList

Deprecated ***demisto-api-**** commands and replaced with ***core-api-**** commands.
##### GetIncidentTasksByState

Deprecated ***demisto-api-**** commands and replaced with ***core-api-**** commands.
```

File: Packs/DemistoRESTAPI/ReleaseNotes/1_3_41.md (hunk 1 above)
Comment (ShirleyDenkberg) [reply Packs/DemistoRESTAPI/ReleaseNotes/1_3_41.md:1]: I reviewed that file.  See my comments.

**Comment by ShirleyDenkberg** [comment]
@DeanArbel @mmhw I reviewed the 1_3_41.md RN file as requested. Doc review completed.

File: Packs/CommonPlaybooks/ReleaseNotes/2_4_40.json [hunk 8]
```diff
@@ -0,0 +1,4 @@
+{
+    "breakingChanges": true,
+    "breakingChangesNotes": "The Demisto REST API integration has been deprecated. Use the Core REST API integration in its place, the functionalities are the same."
```
Comment (ShirleyDenkberg) [inline Packs/CommonPlaybooks/ReleaseNotes/2_4_40.json:3]:
```suggestion
    "breakingChangesNotes": "The Demisto REST API integration has been deprecated. Use the Core REST API integration in its place. The functionalities are the same."
```

File: Packs/CommonPlaybooks/ReleaseNotes/2_4_40.md [hunk 9]
```diff
@@ -0,0 +1,9 @@
+
+#### Playbooks
+
+##### Dedup - Generic v2
+
+Replaced the usage of the deprecated ***Demisto REST API*** integration with ***Core REST API*** integration.
+##### Dedup - Generic v3
+
+Replaced the usage of the deprecated ***Demisto REST API*** integration with ***Core REST API*** integration.
```
Comment (ShirleyDenkberg) [inline Packs/CommonPlaybooks/ReleaseNotes/2_4_40.md:9]:
```suggestion
Replaced the usage of the deprecated **Demisto REST API** integration with the **Core REST API** integration.
##### Dedup - Generic v3

Replaced the usage of the deprecated **Demisto REST API** integration with the **Core REST API** integration.
```

**Comment by ShirleyDenkberg** [comment]
@michal-dagan I re-reviewed the following files as requested:
https://github.com/demisto/content/pull/31388/files#diff-d22cfca11bc16de195a7ecc578d9f1b163709e463185d067c3a29e18fa598439R3
https://github.com/demisto/content/pull/31388/files#diff-d823a30a28710808901760a8bc74447fd27b69363f2f1464b0aabb4ae85d1490
https://github.com/demisto/content/pull/31388/files#diff-9ec365fe2a631516188012bdd10aa50d9d72504866110ebe6ef55ceed1cbe79e

**Review by DeanArbel — APPROVED** [review]
Well done

**Comment by michal-dagan** [comment]
force-merge:
CJ105 - 
Packs/Troubleshoot/Scripts/TroubleshootIsDockerImageExists/TroubleshootIsDockerImageExists.yml - [CJ105]
Packs/DemistoRESTAPI/Scripts/GetIncidentTasksByState/GetIncidentTasksByState.yml - [CJ105]
Packs/Troubleshoot/Scripts/TroubleshootTestInstance/TroubleshootTestInstance.yml - [CJ105]
Packs/ContentManagement/Scripts/JobCreator/JobCreator.yml - [CJ105]
Packs/Troubleshoot/Scripts/TroubleshootGetInstanceParameters/TroubleshootGetInstanceParameters.yml - [CJ105]
GR103 - 
/builds/xsoar/content/Packs/ContentManagement/Playbooks/playbook-Configuration_Setup.yml - [GR103] - opened a ticket:
https://jira-dc.paloaltonetworks.com/browse/CIAC-9211

**Pull request merged**